templates/collection.liquid

```diff
@@ -59,9 +59,29 @@
 
       {% else %}
 
-        <div class="grid__item">
-          <p>{{ 'collections.general.no_matches' | t }}</p>
-        </div>
+        {% if collection.handle == 'all' %}
+
+          {% comment %}
+            Add default products to help with onboarding for collections/all only
+          {% endcomment %}
+
+          {% unless emptyState %}
+            {{ 'theme-onboarding.css' | global_asset_url | stylesheet_tag }}
+            {% assign emptyState = true %}
+          {% endunless %}
+
+          {% include 'onboarding-empty-collection' %}
```

Comment:
I'm a sucker for separate files and includes.  Much easier to read imho.  Easier to remove or turn on and off too.